src/client/Microsoft.Identity.Client/ManagedIdentity/MachineLearningManagedIdentitySource.cs

```diff
@@ -75,7 +75,8 @@ protected override ManagedIdentityRequest CreateRequest(string resource)
             {
                 case AppConfig.ManagedIdentityIdType.ClientId:
                     _requestContext.Logger.Info("[Managed Identity] Adding user assigned client id to the request.");
-                    request.QueryParameters[Constants.ManagedIdentityClientId] = _requestContext.ServiceBundle.Config.ManagedIdentityId.UserAssignedId;
+                    // Use the new 2017 constant for older ML-based environment
+                    request.QueryParameters[Constants.ManagedIdentityClientId2017] = _requestContext.ServiceBundle.Config.ManagedIdentityId.UserAssignedId;
```

Comment:
Didn't we add ML support because someone complained about this? Won't we be breaking them?

Reply:
We added ML support but while doing so broke the source detection order. No one AFAIK used App Service 2017, but if we are adding ML 2017 as a source then it has to be `clientid`